pkg/apis/ext.cattle.io/v1/types.go

```diff
@@ -193,3 +193,58 @@ func (t *Token) GetLastActivitySeen() *metav1.Time {
 func (t *Token) GetCreationTime() metav1.Time {
 	return t.CreationTimestamp
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Kubeconfig contains information about clusters, users, namespaces, and authentication mechanisms.
+type Kubeconfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
Should this note that `Name and GenerateName are not respected. A name is generated with a predefined prefix instead` ?! (Token does)